plugins/experimental/ts_lua/ts_lua_server_request.c

```diff
@@ -258,10 +286,27 @@ ts_lua_server_request_get_headers(lua_State *L)
   while (field_loc) {
     name = TSMimeHdrFieldNameGet(http_ctx->server_request_bufp, http_ctx->server_request_hdrp, field_loc, &name_len);
     if (name && name_len) {
-      value = TSMimeHdrFieldValueStringGet(http_ctx->server_request_bufp, http_ctx->server_request_hdrp, field_loc, -1, &value_len);
       lua_pushlstring(L, name, name_len);
-      lua_pushlstring(L, value, value_len);
-      lua_rawset(L, -3);
+      lua_gettable(L, -2);
+      if (lua_isnil(L, -1)) {
```

Comment:
This definitely needs explanatory comments. I think what is going on is that  the first ``lua_pushlstring`` is used to test whether there is already a value fir this header. If there is you concat to it. But it needs explanations :)